include/net/fota_download.h

```diff
@@ -152,6 +145,31 @@ int fota_download(const char *host, const char *file, const int *sec_tag_list,
 		  uint8_t sec_tag_count, uint8_t pdn_id, size_t fragment_size,
 		  const enum dfu_target_image_type expected_type);
 
+/**@brief Download the given file with the specified image type from the given host.
+ *
+ * Identical to fota_download_start_with_image_type(),
+ * but with additional host configuration options.
+ *
+ * @param host Name of host to start downloading from. Can include scheme
+ *             and port number, for example https://google.com:443
+ * @param file Path to the file you wish to download. See fota_download_any()
+ *             for details on expected format.
+ * @param sec_tag Security tag you want to use with COAPS. Pass -1 to disable DTLS.
+ * @param pdn_id Packet Data Network ID to use for the download, or 0 to use the default.
+ * @param fragment_size Fragment size to be used for the download. If 0, no fragmentation is used.
+ * @param expected_type Type of firmware file to be downloaded and installed.
+ * @param host_cfg Additional host configuration options.
+ *
+ * @retval 0	     If download has started successfully.
+ * @retval -EALREADY If download is already ongoing.
+ * @retval -E2BIG    If sec_tag_count is larger than
+ *		     @kconfig{CONFIG_FOTA_DOWNLOAD_SEC_TAG_LIST_SIZE_MAX}
+ *                   Otherwise, a negative value is returned.
+ */
+int fota_download_with_host_cfg(const char *host, const char *file,
+	int sec_tag, uint8_t pdn_id, size_t fragment_size,
+	const enum dfu_target_image_type expected_type, const struct downloader_host_cfg *host_cfg);
```

Comment:
I wonder if we should define a struct for the host config in `fota_download.h` instead of reusing it from the downloader. On one hand it saves a struct definition and copy of fields when calling the downloader. On the other hand it makes it harder to get the overview of the interface. Now one has to navigate both this header, `downloader.h` and `dfu_target.h` to know what arguments to fill in. Perhaps this is a bigger task than what is in the scope of this PR.